include/internal/libfreenect2/async_packet_processor.h

```diff
@@ -119,6 +135,7 @@ class AsyncPacketProcessor : public PacketProcessor<PacketT>
       {
         // invoke process impl
         processor_->process(current_packet_);
+        releaseBuffer(current_packet_);
```

Comment:
Yes. In theory releaseBuffer should be called as soon as the access to it is finished, but right now no new allocateBuffer will be called before `processor->ready()` becomes true, so putting it in the main loop of async processor is OK.